### PR TITLE
Uglify test (to new branch)

### PR DESCRIPTION
**Pull request by lindsaycarr**
Still need to push out to beta and do some performance tests. This creates a publish function that combines all of the js files into one and then minifies it using `js::uglify_optimize`. Combining means we go from loading 18 individual files to 2 (`all.min.js` and `build_map.js`). Minifying brings the combined file from 98 KB to 54 KB. 

The `build_map.js` script still has to be standalone because it can't be loaded until after divs have been created (I took out some of the functions and put them in a secondary js that was bundled up called `build_map_utils.js`). This is why you see the script tag farther down the page.

Part of #358 and also related to #334.

### CODE REVIEW

**Comment by jread-usgs** [comment]
I see a 20~25% improvement in initial load time w/ this (emulating a 3G connection locally). :+1:

**Comment by jread-usgs** [comment]
Note we also have 6 CSS files that are also taking up multiple connections and likely should be rolled up into one as well.

File: viz.yaml [hunk 1]
```diff
@@ -294,74 +280,22 @@ publish:
...
+      js_files: [
+        "js/url_hashes.js", "js/custom_projection.js", "js/styles.js", "js/buttons.js",
+        "js/resize.js", "js/map_utils.js", "js/states.js", "js/counties.js", "js/watermark.js", 
+        "js/circles.js", "js/mobile_ui.js", "js/modernizr_custom.js", "js/lazy_loading.js", 
+        "js/national_pie.js", "js/rank_states_utils.js", "js/scroll_tracking.js", "js/build_map_utils.js"
+      ]
+    publisher: combine_minify_js
+    scripts: [scripts/publish/combine_minify_js.R]
   -
     id: build_map
```
Comment (jread-usgs) [inline viz.yaml]: I realize this is a small script and I understand the reason why this file is separate from the others, but should `build_map` be minified as well?

Comment (lindsaycarr) [reply viz.yaml]: Oh yeah, I can do that!

**Comment by jread-usgs** [comment]
Interesting that all of the custom js (minus build map) is larger than the state borders topojson: 
image

Neat that the borders are that small.

**Comment by lindsaycarr** [comment]
We really have 6? Huh wow, we do. I'll do some quick research into that separately.

**Comment by jread-usgs** [comment]
yes, but note the 6 includes some (one or two?) that are coming out of the vizlab package.

**Comment by lindsaycarr** [comment]
actually, 4 are from vizlab. This viz only has `custom.css` and `map.css`

**Comment by lindsaycarr** [comment]
OK, merging. Any CSS bundling will come separately.

**Pull request merged**